### PR TITLE
Demo fix for OS X (fix for #437)

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR repairs the 3D demos on OS X.

- Most of the problems were caused by the fact that on OS X, the paintGL function of an OpenGLWidget is called during the setupUi() call (in the MainWindow), when the viewer's scene is still pointing to NULL, which caused a segfault. 
- Some demos did not display 3D effects, this was caused by the fact that on OS X, OpenGL binds automatically the color attribute to 0 in the shaders, and OpenGL does not dipslay anything when a setAttributeValue is made on the attribute 0.
- There were some errors due to Shaders that were still written in GLSL 330 instead of 120.

### CODE REVIEW

File: AABB_tree/demo/AABB_tree/Viewer.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ void Viewer::initializeGL()
 {
   QGLViewer::initializeGL();
   setBackgroundColor(::Qt::white);
-  m_pScene->initGL(this);
+  //m_pScene->initGL(this);
 }
```
Comment (lrineau) [inline AABB_tree/demo/AABB_tree/Viewer.cpp]: Please remove lines, instead of commenting them.

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-105

**Comment by sloriot** [comment]
@MoniqueTeillaud and @odevil: it would be great if you could test the changes so that I can merge them in master if what you reported now works.

File: Polyhedron/demo/Polyhedron/MainWindow.cpp [hunk 2]
```diff
@@ -985,7 +983,8 @@ void MainWindow::open(QString filename)
     default:
       load_pair = File_loader_dialog::getItem(fileinfo.fileName(), selected_items, &ok);
   }
-  
+  viewer->context()->makeCurrent();
+  qDebug()<<"Current";
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/MainWindow.cpp]: should be removed

File: Polyhedron/demo/Polyhedron/Scene.cpp [hunk 3]
```diff
@@ -270,7 +270,7 @@ Scene::duplicate(Item_id index)
 
 void Scene::initializeGL()
 {
-    ms_splatting->init();
+   ms_splatting->init();
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene.cpp]: It seems the diff actually introduces a new indentation incoherence.

File: Polyhedron/demo/Polyhedron/Viewer.cpp [hunk 4]
```diff
@@ -144,7 +144,8 @@ void Viewer::initializeGL()
   vao[0].create();
   for(int i=0; i<3; i++)
     buffers[i].create();
-  d->scene->initializeGL();
+ // qDebug()<<"Viewer's context : "<<context()->contextHandle()->handle();
+ // d->scene->initializeGL();
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Viewer.cpp]: Please remove commented lines.

**Comment by maxGimeno** [comment]
Cleaned-up

File: Mesh_3/demo/Mesh_3/MainWindow.cpp [hunk 5]
```diff
@@ -342,6 +342,7 @@ void MainWindow::open(QString filename)
               io_plugins)
     {
       if(plugin->canLoad()) {
+          viewer->makeCurrent();
         item = plugin->load(fileinfo);
```
Comment (lrineau) [inline Mesh_3/demo/Mesh_3/MainWindow.cpp]: Indentation issue.

**Comment by maxGimeno** [comment]
Cleaned-up

**Comment by sloriot** [comment]
@odevil tested the triangulation and the polyhedron. They worked like a charm.

**Pull request merged**